assets/lokomotive-kubernetes/packet/flatcar-linux/kubernetes/controllers.tf

```diff
@@ -59,6 +59,16 @@ resource "packet_device" "controllers" {
   ipxe_script_url = var.ipxe_script_url
   always_pxe      = false
   tags            = var.tags
+
+  # TODO: review this before merging
+  # depends_on needs a static list expression, so we can't do
+  #   depends_on = var.nodes_depends_on
+  # However, if we add "[]" to the var, it seems to be a static list expression
```

Comment:
`depends_on` must always take a list of objects as far as I know. I guess pointing it to the list will make Terraform wait until all values inside it are "known". I guess this is OK.

Reply:
Yes, but that doesn't explain the tricky part: `var.nodes_depends_on` is already a list. So, actually, this is a list of lists, just so terraform thinks it is a static list. What makes doubt here is that I strongly prefer to not use undocumented features (that may break in new releases)

Reply:
@invidian Do you understand why it workarounds the limitation and if it is a documented terraform feature or not?

Reply:
I think this is a better reference link https://discuss.hashicorp.com/t/tips-howto-implement-module-depends-on-emulation/2305/2 and it seems to me, that this is working as designed (though not explicitly documented).

Honestly, I would call it a workaround. This is just how thing works in Terraform.

Reply:
Saw that too, will maybe update the link to that. Thanks!

I'll sleep on it (and maybe look a little bit terraform code to see the error, not sure) and probably just keep it. Thanks, it helps me to decide to keep it, really :)

Reply:
Simplified the comment and kept it. Thanks =)